File: Scala/src/inl1/RaceTrack.scala
package inl1

import se.lth.cs.window.SimpleWindow

class RaceTrack (val _yStart: Int, val _yFinish: Int) {
  private var yStart = _yStart
  private var yFinish = _yFinish
  
  def draw(w: SimpleWindow) {
    w.setLineWidth(3);
    w.setLineColor(java.awt.Color.RED);
    w.moveTo(40, yFinish);
    w.lineTo(w.getWidth() - 120, yFinish);
    w.setLineColor(java.awt.Color.BLACK);
    w.moveTo(40, yStart);
    w.lineTo(w.getWidth() - 120, yStart);
    w.setLineWidth(1);
  }
  
  def getFinish = yFinish
}

File: inl1/src/inl1/RaceTrack.java
package inl1;

import se.lth.cs.window.SimpleWindow;

public class RaceTrack {

	private int yStart, yFinish;
	
	/** Skapar en kapplöpningsbana.
	 * @param yStart startlinje
	 * @param yFinish mållinje */
	public RaceTrack(int yStart, int yFinish) {
		this.yStart = yStart;
		this.yFinish = yFinish;
	}
	
	/** Ritar kapplöpningsbanan i fönstret w */
	public void draw(SimpleWindow w) {
		int le = (int) (w.getWidth() * 0.025);
		w.setLineWidth(3);
		w.setLineColor(java.awt.Color.RED);
		w.moveTo(le * 2, yFinish);
		w.lineTo(w.getWidth() - le * 3, yFinish);
		w.setLineColor(java.awt.Color.BLACK);
		w.moveTo(le * 2, yStart);
		w.lineTo(w.getWidth() - le * 3, yStart);
		w.setLineWidth(1);
	}
	
	/** Retunerar yFinish */
	int getFinish () {
		return yFinish;
	}
}
